networks/suzuka/suzuka-full-node/src/partial.rs

```diff
@@ -91,69 +90,98 @@ where
-					let movement_transaction = movement_types::Transaction::new(
-						serialized_aptos_transaction,
-						transaction.sequence_number(),
-					);
-					let serialized_transaction = serde_json::to_vec(&movement_transaction)?;
-					transactions.push(BlobWrite { data: serialized_transaction });
+		loop {
+			let remaining = match half_building_time.checked_sub(start.elapsed().as_millis() as u64)
+			{
+				Some(remaining) => remaining,
+				None => {
+					// we have exceeded the half building time
+					break;
 				}
+			};
+
+			match tokio::time::timeout(
+				Duration::from_millis(remaining),
+				self.transaction_receiver.recv(),
+			)
+			.await
+			{
+				Ok(transaction) => match transaction {
+					Ok(transaction) => {
+						info!(
+							target : "movement_timing",
+							batch_id = %batch_id,
+							tx_hash = %transaction.committed_hash(),
+							sender = %transaction.sender(),
+							sequence_number = transaction.sequence_number(),
+							"received transaction",
+						);
+						let serialized_aptos_transaction = serde_json::to_vec(&transaction)?;
+						let movement_transaction = movement_types::Transaction::new(
+							serialized_aptos_transaction,
+							transaction.sequence_number(),
+						);
+						let serialized_transaction = serde_json::to_vec(&movement_transaction)?;
+						transactions.push(BlobWrite { data: serialized_transaction });
+					}
+					Err(_) => {
+						break;
+					}
+				},
 				Err(_) => {
 					break;
 				}
 			}
-
-			if std::time::Instant::now() > end_time {
-				break;
-			}
 		}
 
-		let length = transactions.len();
-		if length > 0 {
-			let mut light_node_client = self.light_node_client.clone();
-			let span = info_span!(
+		if transactions.len() > 0 {
+			info!(
 				target: "movement_timing",
-				"batch_write",
 				batch_id = %batch_id,
-				length = length
+				transaction_count = transactions.len(),
+				"built_batch_write"
 			);
-			light_node_client
-				.batch_write(BatchWriteRequest { blobs: transactions })
-				.instrument(span)
-				.await?;
-			// We now consider the transactions no longer in mempool flight.
-			self.executor.decrement_transactions_in_flight(length as u64);
+			let batch_write = BatchWriteRequest { blobs: transactions };
+			let mut light_node_client = self.light_node_client.clone();
+			tokio::task::spawn(async move {
+				light_node_client.batch_write(batch_write).await?;
+				Ok::<(), anyhow::Error>(())
+			});
 		}
 
 		Ok(())
 	}
 
+	async fn send_transaction_batch_writes(&self) -> Result<(), anyhow::Error> {
+		loop {
+			self.next_transaction_batch_write().await?;
+		}
+	}
+
 	pub async fn write_transactions_to_da(&self) -> Result<(), anyhow::Error> {
 		loop {
-			self.tick_write_transactions_to_da().await?;
+			// run send transaction batch writes and submit transaction batch writes concurrently
+			match self.send_transaction_batch_writes().await {
+				Ok(_) => {}
+				Err(e) => {
+					error!("Failed to send transaction batch writes: {:?}", e);
```

Comment:
Do we want to continue looping at this point, or is it better to return the error?
If the error is not recoverable, this would result in endless spam in the logs.

Reply:
It's not unrecoverable for all failure cases. It could be unrecoverable if the `m1-da-light-node` service goes down.

Reply:
Is there information on the error that we can use to decide if it's recoverable?
We can try to figure the hard way, of course. But I'd rather whitelist the known benign cases than let the node run broken for indefinite amount time.

Reply:
I think we would need to start replacing a bunch of `anyhow::Error`s with typed errors. It almost certainly falls out of scope here. Do you want me to just allow the crash?